_posts/2020-11-12-deconvolution.md

```diff
@@ -0,0 +1,436 @@
+  <line x1="16" y1="6" x2="136" y2="6" />
+  <line x1="23" y1="13" x2="143" y2="13" />
+  <line x1="29" y1="19" x2="149" y2="19" />
+  <line x1="36" y1="26" x2="156" y2="26" />
+  <line x1="42" y1="32" x2="162" y2="32" style="stroke-width:2" />
+
+  <!-- Vertical lines -->
+  <line x1="10" y1="0" x2="42" y2="32" style="stroke-width:2" />
+  <line x1="40" y1="0" x2="72" y2="32" />
+  <line x1="70" y1="0" x2="102" y2="32" />
+  <line x1="100" y1="0" x2="132" y2="32" />
+  <line x1="130" y1="0" x2="162" y2="32" style="stroke-width:2" />
+
+  <!-- Colored Rectangle -->
+  <polygon points="10.0,0.0 130.0,0.0 162.74356617647058,32.74356617647059 42.74356617647059,32.74356617647059" style="fill:#ECB172A0;stroke-width:0"/>
+
+  <!-- Horizontal lines -->
+  <line x1="42" y1="32" x2="162" y2="32" style="stroke-width:2" />
+  <line x1="42" y1="62" x2="162" y2="62" />
+  <line x1="42" y1="92" x2="162" y2="92" />
+  <line x1="42" y1="122" x2="162" y2="122" />
+  <line x1="42" y1="152" x2="162" y2="152" style="stroke-width:2" />
+
+  <!-- Vertical lines -->
+  <line x1="42" y1="32" x2="42" y2="152" style="stroke-width:2" />
+  <line x1="72" y1="32" x2="72" y2="152" />
+  <line x1="102" y1="32" x2="102" y2="152" />
+  <line x1="132" y1="32" x2="132" y2="152" />
+  <line x1="162" y1="32" x2="162" y2="152" style="stroke-width:2" />
+
+  <!-- Colored Rectangle -->
+  <polygon points="42.74356617647059,32.74356617647059 162.74356617647058,32.74356617647059 162.74356617647058,152.74356617647058 42.74356617647059,152.74356617647058" style="fill:#ECB172A0;stroke-width:0"/>
+
+  <!-- Text -->
+  <text x="102.743566" y="172.743566" font-size="1.0rem" font-weight="100" text-anchor="middle" >2048</text>
+  <text x="182.743566" y="92.743566" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(-90,182.743566,92.743566)">2048</text>
+  <text x="16.371783" y="156.371783" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(45,16.371783,156.371783)">950</text>
+</svg>
+</td>
+</tr>
+</table>
+
+
+What we now have is a Dask array composed of 16 CuPy blocks of data.  Notice
+how Dask provides nice typing information in the SVG output.  When we moved
+from NumPy to CuPy, the block diagram above displays `Type: cupy.ndarray` --
+this is a nice sanity check.
+
+The last piece we need before running the deconvolution is the PSF which should
+also be loaded onto the GPU:
+
+```python
+psf = skimage.io.imread("/public/NVMICROSCOPY/PSF.tif")
+c_psf = cp.asarray(psf)
+```
+
+Lastly, we call `map_overlap` with the `deconvolve` function across the Dask
+array:
+
+```python
+out = da.map_overlap(
+    deconvolve,
+    c_imgs,
+    psf=c_psf,
+    iterations=20,
+    meta=c_imgs._meta,
+    depth=tuple(np.array(c_psf.shape) // 2),
+    boundary="periodic"
+)
+out
+```
+
+<a href="/images/deconvolve.png">
+    <img src="/images/deconvolve.png" width="100%"></a>
+
+The image above is taken from a mouse intestine.
```

Comment:
Could you add a little more context? Why is the mouse intestine important and why does the scientist want a deblurred image?

Reply:
I'm not sure I'd agree with these suggestions. For me as a reader, I don't think it's necessary to add any more detail here.

I think it's more or less self-evident why scientists would want a deblurred (i.e. better quality) image. 

Why someone might be choosing to look at a mouse intestine probably depends a lot on what their research question is, which is probably getting off-topic to include in this post.

Reply:
FWIW I've asked my colleagues if they can shed some more light here.

That said, would agree with the points Genevieve has made above.